### PR TITLE
Seleniumbender improvements

### PR DESCRIPTION
**Pull request by tsiknas**
### Changes:
- Added support for config files in seleniumbender Python 
- Setting the platform in capabilities so the tests run in the correct OS when in Saucelabs

### CODE REVIEW

File: bin/config.cfg.template [hunk 1]
```diff
@@ -0,0 +1,31 @@
+[common]
```
Comment (dastergon) [inline bin/config.cfg.template]: We could initialize the keys with a null value or empty strings just for clarity to the key-value format

File: bin/seleniumbender.py [hunk 2]
```diff
@@ -185,21 +198,69 @@ def main():
...
         sys.exit()
 
-    # Read environment variables file
-    output = subprocess.check_output('source {}; env'.format(config), shell=True, executable=SHELL)
-    env_vars = dict((line.split('=', 1) for line in output.splitlines()))
-    os.environ.update(env_vars)
+    if args.saucelabs:
+        os.environ['CODEBENDER_SELENIUM_HUB_URL'] = os.environ['SAUCELABS_HUB_URL']
+    else:
+        os.environ['CODEBENDER_SELENIUM_HUB_URL'] = os.environ['LOCAL_HUB_URL']
```
Comment (dastergon) [inline bin/seleniumbender.py]: default selenium hub url to one of these environs and change the value if the condition does or does not match

File: bin/config.cfg.template (hunk 1 above)
Comment (tsiknas) [inline bin/config.cfg.template]: Fixed at: 4c384a6

File: bin/seleniumbender.py (hunk 2 above)
Comment (tsiknas) [inline bin/seleniumbender.py]: Fixed at: 81c874e

File: tests/walkthrough/test_walkthrough.py [hunk 3]
```diff
@@ -1,11 +1,12 @@
 from selenium.webdriver.common.by import By
 import pytest
 
-from codebender_testing.config import TESTS_USER_AGENT
+from codebender_testing.config import SELENIUM_BROWSER
+from codebender_testing.config import TESTS_USER_AGENT_FIREFOX
+from codebender_testing.config import TESTS_USER_AGENT_CHROME
```
Comment (dastergon) [inline tests/walkthrough/test_walkthrough.py]: imports in lines 4-6 could be merged into a single line with members separated with comma.

```python
from codebender_testing.config import SELENIUM_BROWSER, TESTS_USER_AGENT_FIREFOX, TESTS_USER_AGENT_CHROME
```

File: tests/walkthrough/test_walkthrough.py [hunk 4]
```diff
@@ -1,11 +1,12 @@
...
 from codebender_testing.utils import SeleniumTestCase
 from codebender_testing.utils import SELECT_BOARD_SCRIPT
```
Comment (dastergon) [inline tests/walkthrough/test_walkthrough.py]: same here

File: bin/seleniumbender.py [hunk 5]
```diff
@@ -115,14 +115,16 @@ def noplugin(self, identifier = 'noplugin'):
...
         command = self.create_command('walkthrough', '--plugin')
         retvals = []
-        user_agents = [
-            'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:43.0) Gecko/20100101 Firefox/43.0 codebender-selenium',
-            'Mozilla/5.0 (Windows NT 6.1; rv:43.0) Gecko/20100101 Firefox/43.0 codebender-selenium',
-            'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_1; rv:43.0) Gecko/20100101 Firefox/43.0 codebender-selenium'
-        ]
-        for user_agent in user_agents:
-            retval = self.run_command(command, user_agent=user_agent)
+        for platform in USER_AGENTS.keys():
```
Comment (dastergon) [inline bin/seleniumbender.py]: This one can be implemented the same way as the one I commented in Kokas' PR

File: bin/seleniumbender.py [hunk 6]
```diff
@@ -185,21 +198,68 @@ def main():
...
-    os.environ.update(env_vars)
+    os.environ['CODEBENDER_SELENIUM_HUB_URL'] = os.environ['LOCAL_HUB_URL']
+    if args.saucelabs:
+        os.environ['CODEBENDER_SELENIUM_HUB_URL'] = os.environ['SAUCELABS_HUB_URL']
+
+    capabilities = args.capabilities
+    if capabilities:
+        os.environ['CAPABILITIES'] = capabilities
+
+    # Generate User agents
```
Comment (dastergon) [inline bin/seleniumbender.py]: the User agent generation could be moved to a helper function `generate_user_agents(filename)`

File: bin/seleniumbender.py [hunk 7]
```diff
@@ -185,21 +198,68 @@ def main():
...
+    path = os.path.join(os.path.dirname(__file__), '..', 'codebender_testing', capabilities)
+    with open(path, 'rb') as fp:
+        capabilities_file = yaml.load(fp)
+    for section in capabilities_file:
+        browser = section['browserName']
+        for platform in PLATFORMS.keys():
```
Comment (dastergon) [inline bin/seleniumbender.py]: This one could be slightly modified to be like the one in walktrhough

File: bin/seleniumbender.py [hunk 8]
```diff
@@ -185,21 +198,68 @@ def main():
...
+                user_agent = 'Mozilla/5.0 ({platform}) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/{version}.0.2564.109 Safari/537.36 {identifier}'.format(platform=PLATFORMS[platform], version=version, identifier=USER_AGENT_IDENTIFIER)
+                if platform == 'Linux':
+                    os.environ['SELENIUM_USER_AGENT_CHROME'] = user_agent
+            elif browser == 'firefox':
+                user_agent = 'Mozilla/5.0 ({platform}; rv:{version}.0) Gecko/20100101 Firefox/{version}.0 {identifier}'.format(platform=PLATFORMS[platform], version=version, identifier=USER_AGENT_IDENTIFIER)
+                if platform == 'Linux':
+                    os.environ['SELENIUM_USER_AGENT'] = user_agent
+
+            if not platform in USER_AGENTS:
+                USER_AGENTS[platform] = {}
```
Comment (dastergon) [inline bin/seleniumbender.py]: Try this one instead `USER_AGENTS.setdefault(platform, {})` it does the same thing if it does not find a platform key

File: bin/seleniumbender.py [hunk 9]
```diff
@@ -185,21 +198,68 @@ def main():
...
-    if operation == 'target' and not libraries:
-        print('No target libraries specified!\n')
-        parser.print_help()
+    # Read config file
+    config_parser = ConfigParser.RawConfigParser()
+    config_parser.optionxform = str
+    sections = ['common', target]
+    try:
+        config_parser.read(config)
+        for section in sections:
```
Comment (dastergon) [inline bin/seleniumbender.py]: Check [items()](https://docs.python.org/2/library/configparser.html#ConfigParser.ConfigParser.items)

File: codebender_testing/config.py [hunk 10]
```diff
@@ -99,6 +104,11 @@ def _get_firefox_profile():
         )
     return firefox_profile
 
+def _get_chrome_profile():
+    seed = string.ascii_uppercase + string.ascii_lowercase + string.digits
```
Comment (dastergon) [inline codebender_testing/config.py]: `string.ascii_letters` = `string.ascii_uppercase + string.ascii_lowercase`

File: bin/seleniumbender.py (hunk 5 above)
Comment (tsiknas) [inline bin/seleniumbender.py]: Fixed at: be9fdc389c01648da9b4ad280adc3bca96e17890

File: bin/seleniumbender.py (hunk 6 above)
Comment (tsiknas) [inline bin/seleniumbender.py]: Fixed at: b9dc366

File: bin/seleniumbender.py (hunk 9 above)
Comment (tsiknas) [inline bin/seleniumbender.py]: Used at: 0464664

**Comment by dastergon** [comment]
Looks good to me :+1:

**Pull request merged**